neon-mmd/websurfx
# ♻️ Make the search bar work without `javascript` code

## What does this PR do?

This PR reduces usage of JavaScript on the frontend, instead replacing it with HTML features (using forms for searching) or server-side rendering (for pagination)

## Why is this change important?

This change makes it possible to use basic features of websurfx (searching) without having JavaScript enabled on your browser. It also reduces the amount of scripts that the browser needs to load, making the frontend more lightweight.

## How to test this PR locally?

Pull the changes, build websurfx and run it as usual. Then, open websurfx in your browser and disable JavaScript. Try searching for something, changing parameters (safesearch) and navigating result pages. It should work
